wkcuber/export_wkw_as_tiff.py

```diff
@@ -58,56 +78,95 @@ def wkw_name_and_bbox_to_tiff_name(name: str, slice_index: int) -> str:
     if name is None or name == "":
         return f"{slice_index}.tiff"
     else:
-        return f"name_{slice_index}.tiff"
+        return f"{name}_{slice_index}.tiff"
+
+
+def wkw_slice_to_image(data_slice: np.ndarray):
+    # discard the z dimension
+    data_slice = data_slice.squeeze(axis=3)
+    if data_slice.shape[0] == 1:
+        # discard greyscale dimension
+        data_slice = data_slice.squeeze(axis=0)
+        # swap the axis
+        data_slice = data_slice.transpose((1, 0))
+    else:
+        # swap axis and move the channel axis
+        data_slice = data_slice.transpose((2, 1, 0))
+    return Image.fromarray(data_slice)
 
 
 def export_tiff_slice(
-    export_args: Tuple[int, Tuple[Dict[str, Tuple[int, int, int]], str, str, str]]
+    export_args: Tuple[
+        int,
+        Tuple[
+            Dict[str, Tuple[int, int, int]], str, str, str, Union[None, Tuple[int, int]]
+        ],
+    ]
 ):
-    slice_number, (bbox, dest_path, name, dataset_path) = export_args
-    tiff_bbox = bbox
+    slice_number, (tiff_bbox, dest_path, name, dataset_path, tiling_size) = export_args
     tiff_bbox["topleft"] = [
         tiff_bbox["topleft"][0],
         tiff_bbox["topleft"][1],
         tiff_bbox["topleft"][2] + slice_number,
     ]
     tiff_bbox["size"] = [tiff_bbox["size"][0], tiff_bbox["size"][1], 1]
 
-    tiff_file_name = wkw_name_and_bbox_to_tiff_name(name, slice_number)
+    if tiling_size is None:
+        with wkw.Dataset.open(dataset_path) as dataset:
```

Comment:
You can hoist the with statement to be above the if. Then, you only need it once instead of twice (see line 128).